nova/console/shellinaboxproxy.py

```diff
@@ -13,16 +13,28 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+from nova.compute import rpcapi as compute_rpcapi
 from nova import config
-from nova.consoleauth import rpcapi as consoleauth_rpcapi
+from nova.console.websocketproxy import NovaProxyRequestHandlerBase
 from nova import context
+from nova import exception
 
 STATIC_FILES_EXT = ('.js', '.css', '.html', '.ico', '.png', '.gif')
 
 
-class NovaShellInaBoxProxy(object):
+class NovaShellInaBoxProxy(NovaProxyRequestHandlerBase):
     """Class that injects token validation routine into proxy logic."""
 
+    def __init__(self):
+        self._compute_rpcapi = None
+
+    @property
+    def compute_rpcapi(self):
+        # Lazy load the rpcapi/ComputeAPI upon first use.
```

Comment:
You've taken this from `NovaProxyRequestHandler`, right? Please add a comment saying so, so we can have a look in later upgrades, how things changed there.